extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessRequest.scala

```diff
@@ -72,4 +74,48 @@ object AccessRequest {
     }
     req
   }
+
+  private def getUserGroupsFromUgi(user: UserGroupInformation): util.Set[String] = {
+    user.getGroupNames.toSet.asJava
+  }
+
+  private def getUserGroupsFromUserStore(user: UserGroupInformation): Option[util.Set[String]] = {
+    try {
+      val getUserStoreEnricher = SparkRangerAdminPlugin.getClass.getMethod(
+        "getUserStoreEnricher")
+      getUserStoreEnricher.setAccessible(true)
+      val storeEnricher = getUserStoreEnricher.invoke(SparkRangerAdminPlugin)
+
+      val getRangerUserStore = storeEnricher.getClass.getMethod("getRangerUserStore")
+      getRangerUserStore.setAccessible(true)
+      val userStore = getRangerUserStore.invoke(storeEnricher)
+
+      val getUserGroupMapping = userStore.getClass.getMethod("getUserGroupMapping")
+      getUserGroupMapping.setAccessible(true)
+
+      val userGroupMappingMap: mutable.Map[String, util.Set[String]] =
+        mapAsScalaMap(getUserGroupMapping.invoke(userStore)
+          .asInstanceOf[util.HashMap[String, util.Set[String]]])
+
+      userGroupMappingMap.get(user.getShortUserName)
+    } catch {
+      case _: NoSuchMethodException =>
+        None
+    }
+  }
+
+  private def getUserGroups(user: UserGroupInformation): util.Set[String] = {
+    val enableOverrideUserGroupFromUserStore = SparkRangerAdminPlugin.getRangerConf.getBoolean(
```

Comment:
how about moving this to `SparkRangerAdminPlugin`, so that we can group all possible configurations there from the present to the future

Reply:
Moved to SparkRangerAdminPlugin, and renamed to `ranger.plugin.spark.use.usergroups.from.userstore.enabled`.